pkg/config/structure/unmarshal.go

```diff
@@ -82,7 +66,21 @@ func UnmarshalKey(cfg model.Reader, key string, target interface{}, opts ...Unma
 	}
 
 	if fs.convertArrayToMap {
-		return cfg.UnmarshalKey(key, target, legacyConvertArrayToMap)
+		return cfg.UnmarshalKey(key, target, func(c *mapstructure.DecoderConfig) {
```

Comment:
For readability, could we keep this where it was at the top-level, instead of inlining it? I find it useful to have this func named as a "legacy" feature, plus reducing the indentation here makes it easier to understand.